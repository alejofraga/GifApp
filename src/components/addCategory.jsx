import React, { useState } from 'react';

export const AddCategory = ({ onNewCategory, setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }
    const onSubmitForm = (event) => {
        event.preventDefault();
        const newCategory = inputValue.trim();
        if (newCategory.length < 1) return;
        // enviar el inputValue a gif-app
        onNewCategory(newCategory);
        setInputValue('');
    }
    const resetCategories = () => {
        setCategories([]);

    }

    return (
        <form
            onSubmit={event => onSubmitForm(event)}>
            <input
                type='text'
                placeholder='buscar gif'
                value={inputValue}
                onChange={event => onInputChange(event)}
            >
            </input>
            <i id='reset-btn' onClick={resetCategories} className="fa-solid fa-rotate-right"></i>
        </form>

    )
}

