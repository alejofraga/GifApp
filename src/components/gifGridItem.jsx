import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({ gif }) => {
    return (

        <div className='card'>
            <img src={gif.url} alt={gif.title} key={gif.id}></img>
            <p>{gif.title}</p>
        </div>

    );
}


GifGridItem.propTypes = {
    gif: PropTypes.object.isRequired,
};