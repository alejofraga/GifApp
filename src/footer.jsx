import React from 'react';
import './css/footer.css';

export const Footer = () => {
    return (
        <footer className="pie-pagina">
            <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css" integrity="sha512-DTOQO9RWCH3ppGqcWaEA1BIZOC6xxalwEsw9c2QQeAIftl+Vegovlnee1c9QX4TctnWMn13TZye+giMm8e2LwA==" crossOrigin="anonymous" referrerPolicy="no-referrer" />
            <div className="grupo-1">
                <div className="box">
                    <figure>

                        <img src="images\alejo.jpg" alt="Alejo Fraga" />

                    </figure>
                </div>
                <div className="box">
                    <h2>Alejo Fraga</h2>
                    <p>Estudiante de ingeneria en sistemas en universidad ORT uruguay</p>
                </div>
                <div className="box">
                    <h2>SIGUEME</h2>
                    <div className="red-social">
                        <a href='https://www.linkedin.com/in/alejo-fraga' target="_blank" className="fa-brands fa-linkedin"></a>
                        <a href='https://github.com/alejofraga' target="_blank" className="fa-brands fa-github"></a>
                    </div>
                </div>
            </div>
            <div className="grupo-2">
                <small> 2024 <b>Alejo Fraga</b></small>
            </div>
        </footer>
    );
};
