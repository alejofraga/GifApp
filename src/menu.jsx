import React, { useRef } from 'react';

export const Menu = ({ setCurrentState }) => {



    const hamburgerMenuRef = useRef(null);

    // Función para manejar clics en el menú hamburguesa
    const handleClick = () => {
        // Accedemos al elemento DOM utilizando la referencia y realizamos alguna acción
        hamburgerMenuRef.current.classList.toggle('active');
    };


    const handleClickIndex = () => {
        setCurrentState(1);
    }

    const handleClickTecnologiasUtilizadas = () => {
        setCurrentState(2);
    }

    const handleClickMotivacion = () => {
        setCurrentState(3);
    }


    return (
        <header>
            <div className="container">
                <div onClick={handleClickIndex} className="logo">
                    <img src="images\logo.png" alt="Logo" />
                    <h3>Gif App</h3>
                </div>

                <div className="links">
                    <ul>
                        <li><a onClick={handleClickTecnologiasUtilizadas}>Tecnologías utilizadas</a></li>
                        <li><a onClick={handleClickMotivacion}>Motivación</a></li>
                    </ul>
                </div>

                <div className="overlay"></div>

                <div onClick={handleClick} ref={hamburgerMenuRef} className="hamburger-menu">
                    <div className="bar"></div>
                </div>
            </div>
        </header>
    )
}
