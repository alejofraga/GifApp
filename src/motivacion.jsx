import React from 'react'
import './css/motivacion.css'

export const Motivacion = () => {
  return (
    <div className="motivacion-container">
      <h1>Motivación para desarrollar esta aplicación</h1>
      <p>
        Desarrollar esta aplicación con React ha sido una decisión motivada por diversos factores. Uno de los principales motivos es mi interés en aprender y dominar tecnologías modernas para el desarrollo web. React es una de las bibliotecas de JavaScript más populares y poderosas para la creación de interfaces de usuario interactivas y eficientes.
      </p>
      <p>
        Otra motivación importante es la oportunidad de aplicar y practicar mis habilidades en el desarrollo con React. Crear esta aplicación me permite profundizar en los conceptos y técnicas que he aprendido, así como enfrentarme a desafíos reales que me ayudan a mejorar como desarrollador.
      </p>
      <p>
        Además, el uso de React proporciona ventajas significativas en términos de rendimiento y mantenimiento del código. Su enfoque en componentes reutilizables y su arquitectura basada en el estado facilitan la creación de aplicaciones escalables y fáciles de mantener a largo plazo.
      </p>
      <p>
        En resumen, la motivación detrás del desarrollo de esta aplicación con React radica en mi deseo de aprender, practicar y utilizar tecnologías modernas para crear soluciones web innovadoras y de calidad.
      </p>
    </div>
  );
}
