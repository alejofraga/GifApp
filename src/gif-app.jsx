import React, { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifApp = () => {
    const firstValue = [];
    const [categories, setCategories] = useState(firstValue);
    const onAddCategory = (newCategory) => {
        if (!categories.includes(newCategory)) {
            setCategories([newCategory, ...categories]);
        }

    }

    return (
        <div className='container-grid'>
            <h1>Encuentra tus Gifs favoritos</h1>
            <AddCategory setCategories={setCategories} onNewCategory={newCategory => onAddCategory(newCategory)} />
            {categories.map(category => {

                return <GifGrid category={category} key={category} />
            })
            }
        </div>
    );
};
