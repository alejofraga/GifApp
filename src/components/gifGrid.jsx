import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './gifGridItem';


export const GifGrid = ({ category }) => {

    const { gifList, isLoading } = useFetchGifs(category);
    return (
        <div className='display-grid'>
            {!(isLoading) ? <h3>Se encontraron {gifList.length} gifs relacionados a {category.toUpperCase()}</h3> : null}
            {(isLoading) ? <h4 className='loading'>Cargando...</h4> : null}

            <div className='card-grid' >
                {
                    gifList.map(gif => {
                        return <GifGridItem gif={gif} key={gif.id} />
                    }
                    )
                }
            </div>

        </div>

    );
};
