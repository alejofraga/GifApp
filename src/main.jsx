import React from 'react'
import ReactDOM from 'react-dom/client'
import './css/searchBar.css'
import './css/estilo.css'
import { GifApp } from './gif-app'
import { Menu } from './menu'
import { Footer } from './footer'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <Menu />
    <GifApp />
  </React.StrictMode>,
)
ReactDOM.createRoot(document.querySelector('footer')).render(
  <React.StrictMode>
    <Footer />
  </React.StrictMode>,
)
