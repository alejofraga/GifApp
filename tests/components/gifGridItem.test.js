import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components";


describe('gifGridItem', () => {

    const myGif = {
        id: 1,
        title: 'myGif',
        url: 'http://mygif.com/'
    }

    test('debe hacer match con el snapshot', () => {

        const { container } = render(<GifGridItem gif={myGif} />);

        expect(container).toMatchSnapshot();


    });

    test('debe mostrar la imagen con el url y el alt indicado', () => {

        render(<GifGridItem gif={myGif} />);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(myGif.url);
        expect(alt).toBe(myGif.title);

    });

    test('debe mostrar el titulo en un parrafo', () => {

        render(<GifGridItem gif={myGif} />);

        expect(myGif.title).toBeTruthy();
        
    });


})