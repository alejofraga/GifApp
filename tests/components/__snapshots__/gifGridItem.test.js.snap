// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gifGridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="myGif"
      src="http://mygif.com/"
    />
    <p>
      myGif
    </p>
  </div>
</div>
`;
